timemachine/fe/free_energy.py

```diff
@@ -1181,10 +1221,20 @@ def sample_replica(xvb: CoordsVelBox, state_idx: StateIdx) -> Trajectory:
 
             md_params_replica = replace(md_params, n_frames=1, n_eq_steps=0, seed=state_idx + current_frame)
 
-            return sample_with_context(context, md_params_replica, temperature, ligand_idxs, max_buffer_frames=100)
+            assert md_params_replica.n_frames == 1
+            # Get the next set of frames from the iterator, which will be the only value returned
+            frame, box, final_velos = next(
+                sample_with_context_iter(context, md_params_replica, temperature, ligand_idxs, 1)
```

Comment:
nit: slight preference, for readability
```suggestion
                sample_with_context_iter(context, md_params_replica, temperature, ligand_idxs, max_buffer_frames=1)
```